### PR TITLE
Make NJOBS check cgroups-aware

### PR DESCRIPTION
**Pull request by brianv0**
I haven't fully tested this out with an `eupspkg` mediated built yet.

### CODE REVIEW

**Comment by timj** [comment]
@brianv0 what's the status of this PR? You imply in your comment that you haven't tested it yet.

**Comment by timj** [comment]
@brianv0 this change looks okay to me assuming you tested it in both environments.

**Comment by timj** [comment]
@brianv0 are you happy with this PR as is? Can you rebase?

**Comment by timj** [comment]
@ktlim / @brianv0 shall we merge this? It looks safe in that it only changes behavior if that one file exists.

**Comment by ktlim** [comment]
Seems a bit excessive to list the names of all the CPUs if all you want is a count.  Can't the counting be rolled into the `awk`, avoiding the loop altogether?

**Comment by timj** [comment]
Sounds like we should keep this PR open for now.

**Comment by brianv0** [comment]
In general - the cpuset can look like this:
0-2,4,5,7-10

Where the ranges are always inclusive (so CPUs 2 and 10 are included in the set):

Here is an alternate implementation just doing the counting, tested on centos 7. Still uses awk,  but in a less awk-ward way.

```
total=0
for cpuset in $(cat /sys/fs/cgroup/cpuset/cpuset.cpus | tr ',' '\n'); do
    if [[ $cpuset == *"-"* ]]; then        
        count=$(echo $cpuset | awk -F"-" '{print ($2 + 1) - $1}')
        let "total += $count"
    else
        let "total += 1"
    fi
done
echo $total
```

**Comment by brianv0** [comment]
as a note - this is important when doing builds inside a container on a host where the container has been allocated only some of the CPUs of the host via cgroups. In that case, all CPUs show up in /proc/cpuinfo, but you have to check cgroups to see how many the process was actually allocated.

**Comment by timj** [comment]
If an updated PR arrives that @ktlim approves of I will merge it and make a new release.

File: lib/eupspkg.sh [hunk 1]
```diff
@@ -1170,7 +1170,17 @@ fi
...
 
 if [[ -z ${EUPSPKG_NJOBS} ]]; then
-	NJOBS=$((sysctl -n hw.ncpu || (test -r /proc/cpuinfo && grep processor /proc/cpuinfo | wc -l) || echo 2) 2>/dev/null)   # number of cores on the machine (Darwin & Linux)
+	if [[ -e /sys/fs/cgroup/cpuset/cpuset.cpus ]]; then
+		# Use cgroups information
+		cpulist=()
+		for n in $(cat /sys/fs/cgroup/cpuset/cpuset.cpus | awk '/-/{for (i=$1; i<=$2; i++)printf "%s%s",i,ORS;next} 1' RS=, FS=-); do
+			cpulist+=("$n")
+		done
+		NJOBS="${#cpulist[@]}"
```
Comment (ktlim) [inline lib/eupspkg.sh:1179]:
```suggestion
		NJOBS=$(awk '/-/{i+=$2-$1} {i++} END{print i}' RS=, FS=- /sys/fs/cgroup/cpuset/cpuset.cpus)
```

**Pull request merged**